modules/nextflow/src/main/groovy/nextflow/script/BaseScript.groovy

```diff
@@ -191,9 +191,12 @@ abstract class BaseScript extends Script implements ExecutionContext {
         try {
             run0()
         }
-        catch(InvocationTargetException e) {
+        catch( InvocationTargetException e ) {
             // provide the exception cause which is more informative than InvocationTargetException
-            throw(e.cause ?: e)
+            Throwable target = e
+            while( target.cause )
```

Comment:
I think we should still check `target instanceof InvocationTargetException`, because the target exception could have it's own cause

Reply:
Agree